rasa_nlu/server.py

```diff
@@ -406,6 +408,25 @@ def unload_model(self, request):
             return simplejson.dumps({"error": "{}".format(e)})
 
 
+def conflicting_tokens(token1, token2):
+    return token1 is not None and token2 is not None
+
+
+def get_token():
+    clitoken = cmdline_args.token
+    envtoken = os.environ["RASA_NLU_TOKEN"]
+    if conflicting_tokens(clitoken, envtoken):
+        raise Exception(
+            "RASA_NLU_TOKEN is set both with the -t option,"
+            " with value `{}`, and with and environment variable, "
+            "with value `{}`. "
+            "Please set the token with just one method "
+            "to avoid unexpected behaviours.".format(
+                clitoken, envtoken))
```

Comment:
can we maybe use the command line token by default and only log a warning when the environment token is overwritten?

Reply:
This is very tricky to debug on openshift/kubernets, also if containers have different tokens for NLU the platform won't work properly. I would rather fail early with this than debug the deployment files

Reply:
But don't you see it in the container logs? I mean you have to look there anyway if there is an authentication problem right? And if the there is an exception then the pod will be stopped and the logs are not there anymore

Reply:
The problem is that when you start it with a wrong token it seems to work and in the platform you don't have an error that tells you where the problem is, you get a max recursion exception. This way the problem becomes clear immediately

Reply:
mhm, can we then maybe at least check that we only throw this exception when both tokens are different?

Reply:
I think it will just allow an ambiguous behaviour and that there is no advantage in setting both at the same time. 
I did this thinking about our customers. I want to give them a very simple rule to follow for this, just set the token from the cli or as env variable, not both. Also this is better for as well when debugging.

Reply:
Ok, but then please create an issue and link it to the issue for the next release (if an issue for the next release exists)